dnf/base.py

```diff
@@ -1413,22 +1413,32 @@ def cond_check(pkg):
                 pkg.requires in (pkg.name for pkg in self._goal._installs) or \
                 pkg.requires in [pkg.name for pkg in trans.install]
 
-        def trans_upgrade(query, remove_query):
+        def trans_upgrade(query, remove_query, pkg):
             sltr = dnf.selector.Selector(self.sack)
             sltr.set(pkg=query)
             self._goal.upgrade(select=sltr)
             return remove_query
 
-        def trans_install(query, remove_query):
+        def trans_install(query, remove_query, pkg):
             if self.conf.multilib_policy == "all":
-                self._install_multiarch(query, strict=False)
+                if not pkg.requires:
+                    self._install_multiarch(query, strict=False)
+                else:
+                    # can provide different suggestion for conditional package in comparison to
+                    # "best" policy
+                    if cond_check(pkg):
+                        self._install_multiarch(query, strict=False)
```

Comment:
I see the same result for both conditions: `self._install_multiarch(query, strict=False)`, there should some difference. If the difference is in `cond_check()` function, it should be renamed.